splitio/storage/__init__.py

```diff
@@ -354,4 +354,75 @@ def intersect(self, flag_sets):
         if not isinstance(flag_sets, set) or len(flag_sets) == 0:
             return False
 
-        return any(self.flag_sets.intersection(flag_sets))
\ No newline at end of file
+        return any(self.flag_sets.intersection(flag_sets))
+    
+class RuleBasedSegmentsStorage(object, metaclass=abc.ABCMeta):
+    """SplitRule based segment storage interface implemented as an abstract class."""
+ 
+    @abc.abstractmethod
+    def get(self, segment_name):
+        """
+        Retrieve a rule based segment.
+
+        :param segment_name: Name of the segment to fetch.
+        :type segment_name: str
+
+        :rtype: str
+        """
+        pass
+
+    @abc.abstractmethod
+    def update(self, to_add, to_delete, new_change_number):
+        """
+        Update rule based segment..
+
+        :param to_add: List of rule based segment. to add
+        :type to_add: list[splitio.models.rule_based_segments.RuleBasedSegment]
+        :param to_delete: List of rule based segment. to delete
+        :type to_delete: list[splitio.models.rule_based_segments.RuleBasedSegment]
+        :param new_change_number: New change number.
+        :type new_change_number: int
+        """
+        pass
+
+    @abc.abstractmethod
+    def get_change_number(self):
+        """
+        Retrieve latest rule based segment change number.
+
+        :rtype: int
+        """
+        pass
+
+    @abc.abstractmethod
+    def contains(self, segment_names):
+        """
+        Return whether the traffic type exists in at least one rule based segment in cache.
+
+        :param traffic_type_name: Traffic type to validate.
```

Comment:
I think this doc comment corresponds to a different method, because traffic types are not involved in this method.

Reply:
will fix